### PR TITLE
improve names in transformed openapi schema

### PR DESCRIPTION
**Pull request by jezekra1**

### CODE REVIEW

File: scripts/types_generator/schema_aliases.yaml [hunk 1]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsCreateRequest_moderations
+  - UserRetrieveResponse_result_generate_default_moderations
+  - TextGenerationComparisonCreateRequest_request_moderations
+  - PromptsIdUpdateRequest_moderations
+ModerationsResponse:
+  - PromptsCreateResponse_result_moderations
+  - PromptsIdRetrieveResponse_result_moderations
+  - PromptsIdUpdateResponse_result_moderations
+  - PromptsIdPatchResponse_result_moderations
+  - PromptsRetrieveResponse_results_moderations
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:42]: `ModerationsResponse` -> `ModerationResult`

File: scripts/types_generator/schema_aliases.yaml [hunk 2]
```diff
@@ -0,0 +1,408 @@
...
+ModerationsHAP:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:43]: `ModerationHAPOptions` or maybe just `HAPOptions`

File: scripts/types_generator/schema_aliases.yaml [hunk 3]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsIdUpdateResponse_result_moderations_hap
+  - RequestsChatConversationIdRetrieveResponse_results_request_moderations_hap
+  - TextChatCreateRequest_moderations_hap
+  - TextChatOutputCreateRequest_moderations_hap
+  - TextChatStreamCreateRequest_moderations_hap
+  - TextGenerationComparisonCreateRequest_request_moderations_hap
+  - UserRetrieveResponse_result_generate_default_moderations_hap
+  - UserPatchResponse_result_generate_default_moderations_hap
+  - UserCreateResponse_result_generate_default_moderations_hap
+ModerationsStigma:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:59]: Plural vs Singular

File: scripts/types_generator/schema_aliases.yaml [hunk 4]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsIdUpdateResponse_result_moderations_stigma
+  - RequestsChatConversationIdRetrieveResponse_results_request_moderations_stigma
+  - TextChatCreateRequest_moderations_stigma
+  - TextChatOutputCreateRequest_moderations_stigma
+  - TextChatStreamCreateRequest_moderations_stigma
+  - TextGenerationComparisonCreateRequest_request_moderations_stigma
+  - UserRetrieveResponse_result_generate_default_moderations_stigma
+  - UserPatchResponse_result_generate_default_moderations_stigma
+  - UserCreateResponse_result_generate_default_moderations_stigma
+ModerationsImplicitHate:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:75]: Plural vs Singular

File: scripts/types_generator/schema_aliases.yaml (hunk 2 above)
Comment (Tomas2D) [reply scripts/types_generator/schema_aliases.yaml:43]: same for others, Stigma / Implicit Hate

File: scripts/types_generator/schema_aliases.yaml [hunk 5]
```diff
@@ -0,0 +1,408 @@
...
+FileFormat:
+  - FilesRetrieveResponse_results_file_formats
+  - FilesCreateResponse_result_file_formats
+  - FilesIdRetrieveResponse_result_file_formats
+FilesResult:
+  - FilesRetrieveResponse_results
+  - FilesCreateResponse_result
+  - FilesIdRetrieveResponse_result
+
+# MODERATIONS ------------------------------------------------------------------
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:25]: I would add `Text` prefix

File: scripts/types_generator/schema_aliases.yaml [hunk 6]
```diff
@@ -0,0 +1,408 @@
...
+  - ModelsIdRetrieveResponse_result_tasks
+  - TasksRetrieveResponse_results
+PromptTemplatesResult:
+  - PromptTemplatesRetrieveResponse_results
+  - PromptTemplatesCreateResponse_result
+  - PromptTemplatesIdRetrieveResponse_result
+  - PromptTemplatesIdUpdateResponse_result
+PromptsId:
+  - PromptsRetrieveRequestParamsTaskId
+  - PromptsRetrieveRequestParamsModelId
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:177]: I don't see any class like this.

File: scripts/types_generator/schema_aliases.yaml [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+ParametersReturnOptions:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:178]: `ParametersReturnOptions` -> `GenerationReturnOptions`

File: scripts/types_generator/schema_aliases.yaml [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsIdUpdateResponse_result_parameters_return_options
+  - RequestsChatConversationIdRetrieveResponse_results_request_parameters_return_options
+  - TextChatCreateRequest_parameters_return_options
+  - TextChatOutputCreateRequest_parameters_return_options
+  - TextChatStreamCreateRequest_parameters_return_options
+  - TextGenerationComparisonCreateRequest_request_parameters_return_options
+  - UserRetrieveResponse_result_generate_default_parameters_return_options
+  - UserPatchResponse_result_generate_default_parameters_return_options
+  - UserCreateResponse_result_generate_default_parameters_return_options
+LengthPenalty:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:194]: `LengthPenalty` -> `GenerationLengthPenalty`

File: scripts/types_generator/schema_aliases.yaml [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsIdUpdateResponse_result_parameters_length_penalty
+  - RequestsChatConversationIdRetrieveResponse_results_request_parameters_length_penalty
+  - TextChatCreateRequest_parameters_length_penalty
+  - TextChatOutputCreateRequest_parameters_length_penalty
+  - TextChatStreamCreateRequest_parameters_length_penalty
+  - TextGenerationComparisonCreateRequest_request_parameters_length_penalty
+  - UserRetrieveResponse_result_generate_default_parameters_length_penalty
+  - UserPatchResponse_result_generate_default_parameters_length_penalty
+  - UserCreateResponse_result_generate_default_parameters_length_penalty
+PromptsType:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:210]: `PromptsType` -> `PromptType`

File: scripts/types_generator/schema_aliases.yaml [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsCreateResponse_result_messages_role
+  - PromptsIdRetrieveResponse_result_messages_role
+  - PromptsIdPatchResponse_result_messages_role
+  - PromptsIdUpdateRequest_messages_role
+  - PromptsIdUpdateResponse_result_messages_role
+  - RequestsChatConversationIdRetrieveResponse_results_request_messages_role
+  - TextChatCreateRequest_messages_role
+  - TextChatOutputCreateRequest_messages_role
+  - TextChatStreamCreateRequest_messages_role
+RequestsStatus:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:242]: Singular

File: scripts/types_generator/schema_aliases.yaml [hunk 11]
```diff
@@ -0,0 +1,408 @@
...
+  - TextChatCreateResponse_results_input_tokens_top_tokens
+  - TextChatStreamCreateResponse_results_generated_tokens_top_tokens
+  - TextChatStreamCreateResponse_results_input_tokens_top_tokens
+  - TextGenerationCreateResponse_results_generated_tokens_top_tokens
+  - TextGenerationCreateResponse_results_input_tokens_top_tokens
+  - TextGenerationComparisonCreateResponse_results_result_results_generated_tokens_top_tokens
+  - TextGenerationComparisonCreateResponse_results_result_results_input_tokens_top_tokens
+  - TextGenerationStreamCreateResponse_results_generated_tokens_top_tokens
+  - TextGenerationStreamCreateResponse_results_input_tokens_top_tokens
+BaseTokens:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:267]: What's the difference from GeneratedToken?

File: scripts/types_generator/schema_aliases.yaml [hunk 12]
```diff
@@ -0,0 +1,408 @@
...
+  - TextGenerationStreamCreateResponse_results_generated_tokens
+  - TextGenerationStreamCreateResponse_results_input_tokens
+  - TextChatStreamCreateResponse_results_input_tokens
+  - TextGenerationCreateResponse_results_input_tokens
+  - TextGenerationComparisonCreateResponse_results_result_results_input_tokens
+  - TextChatCreateResponse_results_input_tokens
+  - TextGenerationCreateResponse_results_generated_tokens
+  - RequestsChatConversationIdRetrieveResponse_results_response_results_generated_tokens
+  - TextChatCreateResponse_results_generated_tokens
+TextGenerationParametersLengthPenalty:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:280]: There is one `LengthPenalty`, aren't they same?

File: scripts/types_generator/schema_aliases.yaml [hunk 13]
```diff
@@ -0,0 +1,408 @@
...
+  - TextGenerationCreateRequest_parameters_length_penalty
+  - TextGenerationStreamCreateRequest_parameters_length_penalty
+TextGenerationParametersReturnOptions:
+  - TextGenerationCreateRequest_parameters_return_options
+  - TextGenerationStreamCreateRequest_parameters_return_options
+TextGenerationParametersStopSequences:
+  - TextGenerationCreateRequest_parameters_stop_sequences
+  - TextGenerationStreamCreateRequest_parameters_stop_sequences
+FeedbackCategory:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:289]: `FeedbackCategory` -> `GenerationFeedbackCategory`

File: scripts/types_generator/schema_aliases.yaml [hunk 14]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsCreateRequest_template_data
+  - PromptsIdUpdateRequest_template_data
+  - TextGenerationComparisonCreateRequest_request_template_data
+  - TextGenerationCreateRequest_template_data
+  - TextGenerationStreamCreateRequest_template_data
+  - TextTokenizationCreateRequest_template_data
+  - UserCreateResponse_result_generate_default_template_data
+  - UserPatchResponse_result_generate_default_template_data
+  - UserRetrieveResponse_result_generate_default_template_data
+UserResponseResultGenerateDefault:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:373]: -> `UserGenerationDefault`

File: scripts/types_generator/schema_aliases.yaml [hunk 15]
```diff
@@ -0,0 +1,408 @@
...
+  - PromptsCreateResponse_result
+  - PromptsIdPatchResponse_result
+  - PromptsIdRetrieveResponse_result
+  - PromptsIdUpdateResponse_result
+  - PromptsRetrieveResponse_results
+TextGenerationIdFeedbackResult:
+  - TextGenerationIdFeedbackCreateResponse_result
+  - TextGenerationIdFeedbackRetrieveResponse_result
+  - TextGenerationIdFeedbackUpdateResponse_result
+GenerateParameters:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:390]: We would like to rename every `generate` to `generation`

File: scripts/types_generator/schema_aliases.yaml [hunk 16]
```diff
@@ -0,0 +1,408 @@
...
+  - TextChatOutputCreateRequest_parameters
+  - TextGenerationComparisonCreateRequest_request_parameters
+  - PromptsIdUpdateRequest_parameters
+  - TextChatCreateRequest_parameters
+  - TextChatStreamCreateRequest_parameters
+  - PromptsCreateRequest_parameters
+BaseResponseParameters:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:397]: Why base?

File: scripts/types_generator/schema_aliases.yaml (hunk 12 above)
Comment (jezekra1) [reply scripts/types_generator/schema_aliases.yaml:280]: nope :/
```yaml
    TextGenerationParametersLengthPenalty:
      description: Can be used to exponentially increase the likelihood of the text
        generation terminating once a specified number of tokens have been generated.
      nullable: true
      properties:
        decay_factor:
          description: Represents the factor of exponential decay and must be > 1.0.
            Larger values correspond to more aggressive decay.
          exclusiveMinimum: 1
          nullable: true
          title: Decay factor
          type: number
        start_index:
          description: A number of generated tokens after which this should take effect.
          minimum: 1
          nullable: true
          title: Start index
          type: integer
      title: Length penalty
      type: object
      
    LengthPenalty:
      nullable: true
      properties:
        decay_factor:
          nullable: true
          type: number
        start_index:
          nullable: true
          type: integer
      type: object

```

Comment (Tomas2D) [reply scripts/types_generator/schema_aliases.yaml:280]: Yes please! :]

Comment (jezekra1) [reply scripts/types_generator/schema_aliases.yaml:280]: It's expected behavior, one model is "lean" and the other is from a database

File: scripts/types_generator/schema_aliases.yaml [hunk 17]
```diff
@@ -0,0 +1,406 @@
...
+UserResponseResult:
+  - UserPatchResponse_result
+  - UserRetrieveResponse_result
+PromptsResponseResult:
+  - PromptCreateResponse_result
+  - PromptIdPatchResponse_result
+  - PromptIdRetrieveResponse_result
+  - PromptIdUpdateResponse_result
+  - PromptRetrieveResponse_results
+TextGenerationIdFeedbackResult:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:385]: `TextGenerationFeedbackResult`

File: scripts/types_generator/schema_aliases.yaml [hunk 18]
```diff
@@ -0,0 +1,406 @@
...
+  - TextGenerationIdFeedbackCreateResponse_result
+  - TextGenerationIdFeedbackRetrieveResponse_result
+  - TextGenerationIdFeedbackUpdateResponse_result
+GenerationParameters:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:389]: `TextGenerationParameters`

File: scripts/types_generator/schema_aliases.yaml [hunk 19]
```diff
@@ -0,0 +1,406 @@
...
+  - UserPatchResponse_result_generate_default_template_data
+  - UserRetrieveResponse_result_generate_default_template_data
+UserGenerationDefault:
+  - UserCreateResponse_result_generate_default
+  - UserPatchResponse_result_generate_default
+  - UserRetrieveResponse_result_generate_default
+UserResponseResult:
+  - UserPatchResponse_result
+  - UserRetrieveResponse_result
+PromptsResponseResult:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:379]: Maybe we could remove `Response` sub-string from all names?

File: scripts/types_generator/schema_aliases.yaml [hunk 20]
```diff
@@ -0,0 +1,406 @@
...
+BaseTemplate:
+  - TextGenerationComparisonCreateRequest_request_template
+  - TextTokenizationCreateRequest_template
+  - PromptTemplateOutputCreateRequest_template
+  - UserCreateResponse_result_generate_default_template
+  - UserRetrieveResponse_result_generate_default_template
+  - PromptIdUpdateRequest_template
+  - UserPatchResponse_result_generate_default_template
+  - PromptCreateRequest_template
+TemplateData:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:361]: `PromptTemplateData`

File: scripts/types_generator/schema_aliases.yaml [hunk 21]
```diff
@@ -0,0 +1,406 @@
...
+TextCreateResponseModeration:
+  - TextChatStreamCreateResponse_moderation
+  - TextGenerationStreamCreateResponse_moderation
+GenerationStreamResult:
+  - TextChatStreamCreateResponse_results
+  - TextGenerationStreamCreateResponse_results
+TextGenerationRequestTemplate:
+  - TextGenerationCreateRequest_template
+  - TextGenerationStreamCreateRequest_template
+BaseTemplate:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:352]: `PromptTemplate`

File: scripts/types_generator/schema_aliases.yaml [hunk 22]
```diff
@@ -0,0 +1,406 @@
...
+  - TuneIdRetrieveResponse_result_datapoints
+GenerationResult:
+  - TextGenerationComparisonCreateResponse_results_result_results
+  - TextGenerationCreateResponse_results
+  - TextChatCreateResponse_results
+  - RequestChatConversationIdRetrieveResponse_results_response_results
+TextCreateResponseModeration:
+  - TextChatStreamCreateResponse_moderation
+  - TextGenerationStreamCreateResponse_moderation
+GenerationStreamResult:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:346]: `TextGenerationStreamResult`

File: scripts/types_generator/schema_aliases.yaml [hunk 23]
```diff
@@ -0,0 +1,406 @@
...
+TuneParameters:
+  - TuneCreateRequest_parameters
+  - TuningTypeRetrieveResponse_results_schema_properties_parameters
+TuneResultDatapoints:
+  - TuneImportCreateResponse_result_datapoints
+  - TuneIdPatchResponse_result_datapoints
+  - TuneCreateResponse_result_datapoints
+  - TuneRetrieveResponse_results_datapoints
+  - TuneIdRetrieveResponse_result_datapoints
+GenerationResult:
```
Comment (Tomas2D) [inline scripts/types_generator/schema_aliases.yaml:338]: `TextGenerationStreamResult`

**Review by Tomas2D — COMMENTED** [review]
Add `Text` prefix for all `Generation` properties. In future we can support multiple modalities (audio and so on) which will be part of `generation` module. I would then add type aliases to schema.py for shorter versions.

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**